src/Data/List/Properties.agda

```diff
@@ -792,6 +798,22 @@ module _ (g : B → C) (f : A → Maybe B) where
     mapMaybe (Maybe.map g) (map f xs)  ≡⟨ mapMaybe-map _ f xs ⟩
     mapMaybe (Maybe.map g ∘ f) xs      ∎
 
+mapMaybeIsInj₁∘mapInj₁ : (xs : List A) → mapMaybe (isInj₁ {B = B}) (map inj₁ xs) ≡ xs
+mapMaybeIsInj₁∘mapInj₁ []       = refl
+mapMaybeIsInj₁∘mapInj₁ (x ∷ xs) = cong (x ∷_) (mapMaybeIsInj₁∘mapInj₁ xs)
```

Comment:
Do we have more general versions of this?  Looks like a property of a retract (inj1) aka embedding-projection pair.
So if `proj . emb = just`, then `mapMaybe proj . map emb = id`.

Reply:
You're absolutely right, now generalised.